runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -129,12 +131,16 @@ private void waitForWorkflow(@NonNull UUID workflow, @NonNull ManagementClient c
     public void invoke() {
         for (int x = 0; x < retry; x++) {
             try {
-                runtime.invalidateLayout();
-                Layout requestLayout = new Layout(runtime.getLayoutView().getLayout());
-                ManagementClient orchestrator = getOrchestrator(requestLayout);
-                UUID workflowId = sendRequest(orchestrator);
-                waitForWorkflow(workflowId, orchestrator, timeout, pollPeriod);
-            } catch (NetworkException | TimeoutException e) {
+                Optional<ManagementClient> orchestrator = getOrchestrator();
+                if(orchestrator.isPresent()){
```

Comment:
map + orElseThrow

Reply:
Harder to handle errors this way, better just try to catch them all on this level.

Reply:
This looks cleaner
```
            ManagementClient managementClient = getOrchestrator()
                    .orElseThrow(() -> new IllegalStateException("Orchestrator can not be selected"));

            try {
                UUID workflowId = sendRequest(managementClient);
                waitForWorkflow(workflowId, managementClient, timeout, pollPeriod);
            } catch (NetworkException | TimeoutException | IllegalStateException e) {
                log.warn("WorkflowRequest: Error while running {} on attempt {}, cause {}", this, x, e);
            }
```